ledger/src/blockstore_processor.rs

```diff
@@ -595,9 +595,17 @@ fn process_entries_with_callback(
                     (starting_index..starting_index.saturating_add(transactions.len())).collect()
                 };
 
+                let tx_account_locks_results: Vec<Result<_>> = transactions
+                    .iter()
+                    .map(|tx| tx.get_account_locks(bank.get_transaction_account_lock_limit()))
+                    .collect();
+
```

Comment:
Plan to reuse `tx_account_locks_results` at ln 414, where `calculate_cost(tx)` doesn't need to fetch tx's write locks again. 

Another planned use case is where fee-cache is updated at [here](https://github.com/taozhu-chicago/solana/pull/6/files#diff-281d947e593a76f3c3380804162dc8cf426fcc33e0ae1192589c8193221cd1e2R630-R644)